wireapp/wire-android
# refactor: unify and simplify conversation lists [WPB-9433]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9433" title="WPB-9433" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-9433</a>  [Android] Implement Pagination for Conversation Screen Data Retrieval
  </td></table>
  <br />
 



----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Conversation list composables are overly complicated and have many leftovers after bottom tabs and multiple refactors, so in order to implement pagination, it's better to first make a cleanup.

### Solutions

- removed all legacy code related to bottom tabs
- simplified naming and code of conversation-list-related screens and removed complicated “routers” or “bridges”, now `ConversationsScreenContent` is a composable used to create screens with conversation lists
- removed unneeded `ConversationItemType` from `ConversationRouterState`, extracted it into dedicated file and renamed to `ConversationsDialogsState` as from now on it only contains conversation-list-related dialogs states
- combined `conversationSearchResult` and `foldersWithConversations` into single one as having two separate fields is not necessary and actually would complicate it even more when implementing pagination
- removed duplicated and unused code leftover from previous refactors and updated `ConversationListViewModel` to use `AssistedInject` and thanks to that reduced the number of functions and `LaunchedEffects` needed
- created interfaces for `ConversationListViewModel` and `ConversationCallListViewModel` and “_Preview_” versions of them to be able to create previews for screens that show conversation list and use `ConversationsScreenContent`
- added previews for conversation list composable, all conversations screen and archive screen
- extracted empty screens to dedicated files (search, archive and all conversations)
- changed parameter from `Activity` to `Context` for functions that return call intents, to be able to show previews (there is no `Activity` for composable previews)
- updated tests, refactored to have `Arrangement` classes

### Testing

#### Test Coverage (Optional)


#### How to Test

Open all conversations list or archive list and make sure everything works as it should.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
